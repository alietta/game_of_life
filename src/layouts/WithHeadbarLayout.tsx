import React from 'react';
import { Container, useTheme } from 'sancho';
import { Headbar } from '@/modules/Headbar';
import { Layout } from './LayoutStyles';

export const WithHeadbarLayout: React.FC = (props) => {
  const theme = useTheme();
  return (
    <Layout background={theme.colors.background.default}>
      <Headbar />
      <Container css={{ maxWidth: 1024, paddingTop: 50 }}>
        {props.children}
      </Container>
    </Layout>
  );
};
