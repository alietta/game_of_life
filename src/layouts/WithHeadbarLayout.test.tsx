import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { matchers } from 'jest-emotion';
import styled from '@emotion/styled';
import { WithHeadbarLayout } from './WithHeadbarLayout';

expect.extend(matchers);

const mockStore = configureMockStore([thunk]);
const store = mockStore({
  user: {
    name: 'Helen',
  },
});
describe('WithHeadbarLayout', () => {
  const Component: React.FC = () => <h1>Test</h1>;
  it('render', () => {
    const element = mount(
      <WithHeadbarLayout>
        <Component />
      </WithHeadbarLayout>,
      {
        wrappingComponent: Provider,
        wrappingComponentProps: {
          store,
        },
      }
    );
    expect(element.html()).toMatchSnapshot();
  });
  it('render with fallback background', () => {
    jest.mock('sancho', () => ({
      Container: () => <div name="container" />,
      IconLogOut: () => <div />,
      Button: () => <div />,
      IconSun: () => <div />,
      useTheme: () => ({ colors: { background: { default: undefined } } }),
    }));
    const element = mount(
      <WithHeadbarLayout>
        <Component />
      </WithHeadbarLayout>,
      {
        wrappingComponent: Provider,
        wrappingComponentProps: {
          store,
        },
      }
    );
    expect(element.find('Layout')).toHaveStyleRule('background', 'white');
  });
});
