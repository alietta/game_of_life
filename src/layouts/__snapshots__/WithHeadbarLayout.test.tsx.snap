// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WithHeadbarLayout render 1`] = `"<div class=\\"css-cqp92b-WithHeadbarLayout\\"><div class=\\"css-1uom1gm-Header\\"><div class=\\"css-14et220-Header\\"><div class=\\"Button Touchable css-t8zin7\\" role=\\"button\\" tabindex=\\"0\\" name=\\"theme\\"><svg viewBox=\\"0 0 24 24\\" fill=\\"none\\" stroke=\\"#212529\\" stroke-width=\\"2\\" stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" height=\\"20px\\" width=\\"20px\\"><circle cx=\\"12\\" cy=\\"12\\" r=\\"5\\"></circle><line x1=\\"12\\" y1=\\"1\\" x2=\\"12\\" y2=\\"3\\"></line><line x1=\\"12\\" y1=\\"21\\" x2=\\"12\\" y2=\\"23\\"></line><line x1=\\"4.22\\" y1=\\"4.22\\" x2=\\"5.64\\" y2=\\"5.64\\"></line><line x1=\\"18.36\\" y1=\\"18.36\\" x2=\\"19.78\\" y2=\\"19.78\\"></line><line x1=\\"1\\" y1=\\"12\\" x2=\\"3\\" y2=\\"12\\"></line><line x1=\\"21\\" y1=\\"12\\" x2=\\"23\\" y2=\\"12\\"></line><line x1=\\"4.22\\" y1=\\"19.78\\" x2=\\"5.64\\" y2=\\"18.36\\"></line><line x1=\\"18.36\\" y1=\\"5.64\\" x2=\\"19.78\\" y2=\\"4.22\\"></line></svg></div><span class=\\"css-1s0denh-Header\\">Helen</span><div class=\\"Button Touchable css-t8zin7\\" role=\\"button\\" tabindex=\\"0\\" name=\\"logout\\"><svg viewBox=\\"0 0 24 24\\" fill=\\"none\\" stroke=\\"#212529\\" stroke-width=\\"2\\" stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" height=\\"20px\\" width=\\"20px\\"><path d=\\"M9 21H5a2 2 0 0 1-2-2V5a2 2 0 0 1 2-2h4\\"></path><polyline points=\\"16 17 21 12 16 7\\"></polyline><line x1=\\"21\\" y1=\\"12\\" x2=\\"9\\" y2=\\"12\\"></line></svg></div></div></div><div class=\\"css-1uhxri2-WithHeadbarLayout\\"><h1>Test</h1></div></div>"`;
